### ISSUE TITLE
Update function does not override folders in release

### ISSUE THREAD

**Issue by Yarden-zamir** [ATLauncher/ATLauncher#862]
### What happened?

The update function seems to not override the folders but instead to combine them with the old ones, this differs from every other launcher and is very problematic. Our modpack users that use ATLauncher will get a broken build if we ever remove/rename or even move a script, this is a really big issue for us because of that.

Example: we renamed the script remove_crafting_table.js to devices.js and a user had duplicated recipes.

### Operating System

Windows

### Is this actually a launcher bug?

- [X] I am sure this is a bug with the launcher and not an issue with running Minecraft

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by RyanTheAllmighty** [comment]
Yeah this is a relic of the past when ATLauncher packs had specific commands in order to fix this issue.

Since we only delete the configs folder, and nothing else.

This will likely be solved as part of #704.

I didn't think any other launchers supported updating modpacks/instances, but will have to confirm with what others are doing if that's the case that they don't have issues on other launchers.

But I imagine we either snapshot what files exist in the instance folder (part of the solution for #704 as well) and then just nuke all of those files, but also presents issues on update as there are now other files and saves.

Super important not to delete anything the user may want/need to run the instance, so it's a super tricky one.

**Comment by Yarden-zamir** [comment]
Other launchers don't have this issue. Not sure their approach but I think that, especially after the change to server-configs and such, deletion of config folder and others is fine because user domain stuff is in world or in options.txt and that's it.

I think a whitelist of what not to delete is the correct approach
